### DISCUSSION TITLE
Pricing for wallet integration

### DISCUSSION THREAD

**Discussion by EthanHuntIMF99** [rainbow-me/rainbowkit#2001] [Q&A] [answered]
Dear RainbowKit Team,

We are in the process of developing  decentralized application (dApp) and are very interested in integrating RainbowKit for wallet management and user authentication.

We have been thoroughly impressed by the features and flexibility that RainbowKit offers and believe it would be an excellent fit for our project. However, we would like to get a detailed understanding of the pricing structure associated with using RainbowKit in our application. Specifically, we are interested in any costs related to:

1. Basic integration and usage fees
2. Fees associated with high-volume transactions and active users
3. Any additional costs for premium features or customizations
4. How is an active wallet tracked?(Is a transaction an active wallet or is login also considered an active wallet?)

We aim to ensure a seamless and efficient wallet experience for our users and would appreciate any information you can provide regarding your pricing plans.

Thank you for your time and assistance. We look forward to the possibility of working with RainbowKit to enhance our gaming dApp.

Ps: While we know its free to use but we are talking huge numbers here as the time progresses.

**Comment by DanielSinclair** [comment, marked as answer]
RainbowKit is entirely free and open-source. You will need to use a node provider with Wagmi for optimal usage and uptime for your dApp, so reference a provider like Alchemy for pricing there. It sounds like you're also potentially referring to Account Abstraction pricing models. This is separate from RainbowKit, but you can reference providers like Alchemy AccountKit or ZeroDev to learn about integrating those smart contract accounts for your users, and subsidizing the cost of gas for user transactions with a paymaster and bundler.
